### PR TITLE
Handle systems where `hostname` is unavailable

### LINKED ISSUE #20758: "could not spawn `hostname`: no such file or directory " libgit2 test failure when building RPM nightlies
This test failure appeared over the last few days when building the RPM nightlies. It happens reliably on both 64-bit and 32-bit, and with libgit2 0.25 and 0.24. I couldn't find any obvious suspects in the recent PRs.
```
Error in testset libgit2:
Error During Test
  Got an exception of type Base.UVError outside of a @test
  could not spawn `hostname`: no such file or directory (ENOENT)
  Stacktrace:
   [1] lookup_ref at ./distributed/remotecall.jl:151 [inlined]
   [2] lookup_ref at ./distributed/remotecall.jl:149 [inlined] (repeats 2 times)
   [3] #remotecall_fetch#131(::Array{Any,1}, ::Function, ::Function, ::Base.Distributed.Worker, ::String, ::Vararg{String,N} where N) at ./distributed/remotecall.jl:347
   [4] #remotecall_fetch#131(::Array{Any,1}, ::Function, ::Function, ::Base.Distributed.Worker, ::String, ::Vararg{String,N} where N) at ./distributed/remotecall.jl:350
   [5] _setindex!(::Dict{Int32,String}, ::String, ::Int32, ::Int32) at ./dict.jl:407
   [6] _setindex!(::Dict{Int32,String}, ::String, ::Int32, ::Int32) at ./dict.jl:394
ERROR: LoadError: Test run finished with errors
while loading /builddir/build/BUILD/julia/test/runtests.jl, in expression starting on line 38
```
https://copr-be.cloud.fedoraproject.org/results/nalimilan/julia-nightlies/fedora-25-i386/00516542-julia/build.log.gz

### PR DESCRIPTION
**Pull request by omus**
Fixes #20758

### CODE REVIEW

**Comment by tkelman** [comment]
was the julia gethostname function not working?

**Comment by omus** [comment]
Ugh, I searched for `Base.host` and `Base.hostname`. Looking at the `gethostname` source it looks like we can still run into exceptions so I'll leave the `try` block.

**Comment by tkelman** [comment]
xref #13715 and #20371

File: test/libgit2.jl [hunk 1]
```diff
@@ -1013,18 +1013,22 @@ mktempdir() do dir
             end
 
             # Find a hostname that maps to the loopback address
-            hostname = replace(readchomp(`hostname`), r"\..*$", "")
-            loopback = ip"127.0.0.1"
+            hostnames = ["localhost"]
+            try
+                # In minimal environments a hostname might not be available (issue #20758)
+                unshift!(hostnames, replace(gethostname(), r"\..*$", ""))
```
Comment (tkelman) [inline test/libgit2.jl]: why is it this is stripping anything after the last dot? (if I'm reading that regex right)

Comment (omus) [inline test/libgit2.jl]: The regex strips everything after the first dot ("a.b.c" -> "a"). Some environment (Macs) have "macbook.local" bound to the external address while "macbook" is bound to the loopback address.

Comment (tkelman) [inline test/libgit2.jl]: worth a comment on that?

Comment (omus) [inline test/libgit2.jl]: Will do

File: test/libgit2.jl [hunk 2]
```diff
@@ -1014,8 +1014,11 @@ mktempdir() do dir
 
             # Find a hostname that maps to the loopback address
             hostnames = ["localhost"]
+
+            # In minimal environments a hostname might not be available (issue #20758)
             try
-                # In minimal environments a hostname might not be available (issue #20758)
+                # Some environments, namely Macs, the hostname "macbook.local" bound to the
```
Comment (tkelman) [inline test/libgit2.jl]: In some environments ?
...
is bound to

Comment (StefanKarpinski) [inline test/libgit2.jl]: I think this is clear enough. Comments in internal files don't need to be 100% grammatically perfect.

Comment (omus) [inline test/libgit2.jl]: I'll change it. The "is bound to" mistake is going to bother me

**Pull request merged**